File: submitted code/58.length-of-last-word.java
//
// @lc app=leetcode.cn id=58 lang=java
//
// [58] length-of-last-word
//
class Solution {
    public int lengthOfLastWord(String s) {
        int l = s.length();
        int i = l-1;
        while(s.charAt(i) == ' ')
            --i;
        int ans = 0;
        for(; i >= 0; --i){
            if(s.charAt(i) == ' ')
                break;
            ++ans;
        }
        return ans;
    }
}
// @lc code=end

File: submitted_code/58.length-of-last-word.cpp
//
// @lc app=leetcode.cn id=58 lang=cpp
//
// [58] length-of-last-word
//
class Solution {
public:
    int lengthOfLastWord(string s) {
        int tmp=0;
        int i=s.length()-1;
        while(i > -1 && s[i] == ' ')
            --i;
        while(i > -1 && s[i] != ' '){
            ++tmp;
            --i;
        }
        return tmp;
    }
};
// @lc code=end